dessia_common/datatools/modeling.py

```diff
@@ -0,0 +1,734 @@
+"""
+Tools and base classes for machine learning methods.
+
+"""
+from typing import List, Dict, Any
+
+import numpy as npy
+from sklearn import preprocessing, linear_model, ensemble, tree, svm, neural_network
+
+from dessia_common.core import DessiaObject
+
+
+# ======================================================================================================================
+#                                                     S C A L E R S
+# ======================================================================================================================
+class BaseScaler(DessiaObject):
+    _rebuild_attributes = []
+
+    def __init__(self, name: str = ''):
+        DessiaObject.__init__(self, name=name)
+
+    @classmethod
+    def _skl_class(cls):
+        raise NotImplementedError('Method _skl_class not implemented for BaseScaler. Please use children.')
+
+    def _call_skl_scaler(self):
+        return self._skl_class()()
+
+    def instantiate_skl(self):
+        scaler = self._call_skl_scaler()
+        for attr in self._rebuild_attributes:
+            setattr(scaler, attr, getattr(self, attr))
+        return scaler
+
+    @classmethod
+    def instantiate_dessia(cls, scaler, name: str = ''):
+        kwargs = {'name': name}
+        for attr in cls._rebuild_attributes:
```

Comment:
In the lights of our discussion on slack, consider the following : 

```
kwargs = {a: get_scaler_attr(scaler, a) for a in cls._rebuild_attributes}
kwargs["name"] = name
return cls(**kwargs)
```

with the following get_scaler_attr_function : 
```
def get_scaler_attr(scaler, attr):
    scaler_attr = getattr(scaler, attr)
    if isinstance(scaler_attr, nps.ndarray):
        return scaler_attr.tolist()
    return scaler_attr
```

Reply:
In any case, I think getattr(scaler, attr) should be stored in a variable to avoid calling getattr three times

Reply:
So you advise me to code a specific function outside scalers ?

Reply:
It's done.